From: Simon McVittie <smcv@debian.org>
Date: Mon, 18 Mar 2024 10:54:02 +0000
Subject: girparser: Make sizes in integer_aliases more obviously correct

We don't actually need to use the Meson-detected size macros here,
because the result of `sizeof()` is an integer constant expression.

No functional change. Equivalent to
https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3970 in GLib.

Signed-off-by: Simon McVittie <smcv@debian.org>
Applied-upstream: 1.80.1, commit:b7d2e77d56766d12b08069b0ca4d32b11e0225bc
---
 girepository/girparser.c | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/girepository/girparser.c b/girepository/girparser.c
index f63136c..80fc270 100644
--- a/girepository/girparser.c
+++ b/girepository/girparser.c
@@ -437,16 +437,16 @@ typedef struct {
 } IntegerAliasInfo;
 
 static IntegerAliasInfo integer_aliases[] = {
-  { "gchar",    SIZEOF_CHAR,      1 },
-  { "guchar",   SIZEOF_CHAR,      0 },
-  { "gshort",   SIZEOF_SHORT,     1 },
-  { "gushort",  SIZEOF_SHORT,     0 },
-  { "gint",     SIZEOF_INT,       1 },
-  { "guint",    SIZEOF_INT,       0 },
-  { "glong",    SIZEOF_LONG,      1 },
-  { "gulong",   SIZEOF_LONG,      0 },
-  { "gssize",   GLIB_SIZEOF_SIZE_T,    1 },
-  { "gsize",    GLIB_SIZEOF_SIZE_T,    0 },
+  { "gchar",    sizeof (gchar),   1 },
+  { "guchar",   sizeof (guchar),  0 },
+  { "gshort",   sizeof (gshort),  1 },
+  { "gushort",  sizeof (gushort), 0 },
+  { "gint",     sizeof (gint),    1 },
+  { "guint",    sizeof (guint),   0 },
+  { "glong",    sizeof (glong),   1 },
+  { "gulong",   sizeof (gulong),  0 },
+  { "gssize",   sizeof (gssize),  1 },
+  { "gsize",    sizeof (gsize),   0 },
   { "gintptr",  sizeof (gintptr),      1 },
   { "guintptr", sizeof (guintptr),     0 },
 };
